src/pybind/mgr/mgr_module.py

```diff
@@ -324,15 +324,20 @@ def load_func_metadata(f: HandlerFuncType) -> Tuple[str, Dict[str, Any], int, st
         if full_argspec.defaults:
             first_default -= len(full_argspec.defaults)
         args = []
+        positional = True
         for index, arg in enumerate(full_argspec.args):
             if arg in CLICommand.KNOWN_ARGS:
                 continue
+            if arg == '_end_positional_':
```

Comment:
instead of introducing a convention for defining command signature, i'd suggest use a notation like:

```py
def do_command(self, arg: str, opt_: str):
  """
  do something
  """
  # ...
```

or even defining types for non-positional arguments:

```py
from typing import NewType

# poor man's generic type
class CephOptionTypes:
  def __init__(self):
    self.types = {}
  
  def __getitem__(self, tp):
    option_tp = self.types.get(tp)
    if option_tp is None:
      option_tp = NewType("CephOpt"+str(tp), tp)
      self.types[tp] = option_tp
    return option_tp

  def get_inner(self, tp):
    for orig, new_type in self.types.items():
      if tp is new_type:
        return orig
    return None

OptionTypes = CephOptionTypes()

def do_command(self, arg: str, opt: OptionTypes[str]):
  """
  do something
  """
  # ...


def _compound_type_to_argdesc(tp, attrs):
  # ...
  else:
    inner_tp = OptionTypes.get_inner(tp)
    if inner_tp is not None:
      return CephArgtype.to_argdesc(inner_tp, attrs, positional=False)
  raise ValueError(f"unknown type '{tp}': '{attrs}'")
```

Reply:
after a second thought, i think a separator is better despite that it's a "magic number". as it's less error-prune.